mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/Activator.java

```diff
@@ -18,24 +18,44 @@
 
 import com.mucommander.snapshot.MuSnapshot;
 import com.mucommander.viewer.FileViewerService;
-import javax.annotation.ParametersAreNonnullByDefault;
+import com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReaderSpi;
+import com.twelvemonkeys.imageio.plugins.psd.PSDImageReaderSpi;
+import com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReaderSpi;
+import com.twelvemonkeys.imageio.plugins.webp.WebPImageReaderSpi;
+
 import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.imageio.spi.IIORegistry;
 
 /**
  * Activator for viewer for image files.
  */
 @ParametersAreNonnullByDefault
 public class Activator implements BundleActivator {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(Activator.class);
+
     private ServiceRegistration<FileViewerService> viewerRegistration;
 
     @Override
     public void start(BundleContext context) throws Exception {
         MuSnapshot.registerHandler(new ImageViewerSnapshot());
-
-        viewerRegistration = context.registerService(FileViewerService.class, new ImageFileViewerService(), null);
+        try {
+            IIORegistry registry = IIORegistry.getDefaultInstance();
+            registry.registerServiceProvider(new JPEGImageReaderSpi());
+            registry.registerServiceProvider(new PSDImageReaderSpi());
+            registry.registerServiceProvider(new TIFFImageReaderSpi());
+            registry.registerServiceProvider(new WebPImageReaderSpi());
+        } catch (Exception e) {
+            LOGGER.error("Error registering additional image service providers", e);
```

Comment:
shall we exit in this case?

Reply:
Noe, and I'm not sure if it can throw any exception at this point. This is rather to be on safe-side and continue to work. If some of those providers won't register itself, then still be the built-in ones (in JRE), and since the list of files that can be previewed is taken dynamically the Preview will still work (for some of files).

Reply:
we can raise an exception here but yeah, if the bundle doesn't start we won't have any support of images - so let's keep it this way